git-daemon: add page (#5938)
tldr-pages/tldr: pages/common/git-daemon.md

# git daemon

> A really simple server for Git repositories.
> More information: <https://git-scm.com/docs/git-daemon>.

> - Git daemon with a whitelisted set of directories:

Reviewer: Perhaps this will read better?
```suggestion
- Launch a Git daemon with a whitelisted set of directories:
```

`git daemon --export-all {{path/to/directory1}} {{path/to/directory2}}`

> - Git daemon with a base directory and allow pulling from all directories that look like Git repos:

Reviewer: Perhaps this will read better?
```suggestion
- Launch a Git daemon with a base directory and allow pulling from all directories that look like Git repos:
```

> - Launch a Git daemon with a base directory and allow pulling from all directories that look like Git repos:

Reviewer 2: ```suggestion
- Launch a Git daemon with a base directory and allow pulling from all directories that look like Git repositories:
```
Does the first example have no base directory? If it does, I'd say `with a specific base directory`

`git daemon --base-path={{path/to/directory}} --export-all --reuseaddr`

> - Git daemon with a base directory with allowed push access and verbosity:

Reviewer 3: ```suggestion
- Launch a Git daemon with a base directory with allowed push access and verbosity:
```

> `git daemon --base-path={{path/to/directory}} --export-all --enable=receive-pack --reuseaddr --informative-errors --verbose`

Reviewer: This seems like a lot to pack into a single example. There's plenty of space left on this page, so maybe we could split it up into multiple examples?

Author: git-daemon doesn't have much options (it's one of those things that nobody will ever use), so I think it's fine as it is

Reviewer 3: Maybe you should move `--informative-errors --verbose` to another example about verbosity.  This example looks _unnecessarily_ ugly.

Reviewer 3: Nevermind; I see you've documented the other flags previously.

Reviewer: If a command doesn't have many options, that's a great reason to split them up into multiple examples. Remember, in tldr pages the goal is clarity and readability to assist readers in understanding how to use the command in question with the lowest possible cognitive load.

> - Launch a Git daemon with a base directory with allowed push access and verbosity:
> 
> `git daemon --base-path={{path/to/directory}} --export-all --enable=receive-pack --reuseaddr --informative-errors --verbose`

Reviewer 3: wdyt @marchersimon?
```suggestion
- Launch a Git daemon, verbosely printing log messages and allowing Git clients to push:

`git daemon --enable=receive-pack --informative-errors --verbose`
```

Reviewer 2: Sounds good, however `push` alone sounds a bit weird. How about ` allowing Git clients to push commits`? (On the other side you can push more than just commits, right?)

Reviewer 3: Yes, you can push more than commits — you can also push tags and branches.
